cmd/pj-rehearse-metrics/pj-rehearse-metrics.go

```diff
@@ -0,0 +1,172 @@
+package main
+
+import (
+	"cloud.google.com/go/storage"
+	"context"
+	"flag"
+	"fmt"
+	"google.golang.org/api/iterator"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"sync"
+)
+
+var (
+	originCiBucket   = "origin-ci-test"
+	rehearseJobName  = "pull-ci-openshift-release-master-pj-rehearse"
+	metricsPathInRun = "artifacts/rehearse-metrics.json"
+)
+
+type scraper struct {
+	cacheDir string
+	bucket   *storage.BucketHandle
+	ctx      context.Context
+}
+
+func (s *scraper) scrapeRun(runPrefix string, prNumber string) {
+	runNumber := path.Base(runPrefix)
+	metricsPath := path.Join(runPrefix, metricsPathInRun)
+
+	prDir := filepath.Join(s.cacheDir, prNumber)
+	localMetricsPath := filepath.Join(prDir, runNumber)
+	if _, err := os.Stat(localMetricsPath); err == nil {
+		return
+	}
+
+	reader, err := s.bucket.Object(metricsPath).NewReader(s.ctx)
+	if err != nil {
+		return
+	}
+	data, err := ioutil.ReadAll(reader)
+	_ = reader.Close()
+	if err != nil {
+		fmt.Printf("could not read rehearse metrics err=%v\n", err)
+		return
+	}
+
+	if err := os.MkdirAll(prDir, 0755); err != nil {
+		fmt.Printf("could not create a directory for PR err=%v\n", err)
+		return
+	}
+
+	if err := ioutil.WriteFile(localMetricsPath, data, 0644); err != nil {
+		fmt.Printf("could not write metrics to %s err=%v\n", localMetricsPath, err)
+		return
+	}
+}
+
+func (s *scraper) scrapePr(prefix string) {
+	prNumber := path.Base(prefix)
+	jobPrefix := path.Join(prefix, rehearseJobName)
+	jobPrefix = fmt.Sprintf("%s/", jobPrefix) // Needed because path.Join() removes trailing slash
```

Comment:
yeah technically you should not use `path.Join` for URLs but whatever

Reply:
I think I'll go with the `path.Join` way, the object names behave more like files than URLs, so `path` operations are probably a bit more readable, too. The only problem is that the separator is a part of the object name, not an actual separator.